src/components/categories/EditCategoriesModal.tsx

```diff
@@ -0,0 +1,113 @@
+import AddIcon from "@mui/icons-material/Add";
+import EditIcon from "@mui/icons-material/Edit";
+import { Button, Modal } from "@mui/material";
+import { DataStore } from "aws-amplify";
+import { useEffect, useState } from "react";
+import { Category, ItemCategory, LazyCategory } from "../../models";
+import "../../styles/EditCategoriesModal.css";
+import EditCategoryChildModal from "./EditCategoryChildModal";
+import EditTag from "./EditTag";
+
+function EditCategoriesModal() {
+  const [categories, setCategories] = useState<LazyCategory[]>([]);
+  const [open, setOpen] = useState(false);
+  const [addOpen, setAddOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  useEffect(() => {
+    const fetchCategories = async () => {
+      const cats = await DataStore.query(Category);
+      setCategories(cats);
+    };
+    fetchCategories();
+  }, []);
+
+  const handleDelete = async (id: string) => {
+    await DataStore.delete(Category, (c) => c.id.eq(id));
+    await DataStore.delete(ItemCategory, (ic) => ic.categoryId.eq(id));
```

Comment:
Should we allow user to delete a category? Since all the users are sharing the same set of category, if one user deleted a category and we removed all the related ItemCategory, it will affect the item from other users too.

Reply:
I think we should let them do it and I do agree it is a very destructive action. Maybe when the user clicks on the delete button for a tag, a warning/confirmation modal shows up to tell them what they are about to do?

Reply:
That's a good idea. Will we add the warning / confirmation modal for this issue or create a new issue for it?

Reply:
I can do it in this issue. I imagine it would just pop up after clicking delete. Something like this:
![image](https://github.com/hack4impact-calpoly/wilshire-hope-chest/assets/77358521/71963da8-ca93-4097-8699-9fde0c7c15f4)

Reply:
Yes, that looks good to me! the wrong message can be something like "Do you really want to delete this category? This process cannot be undone and apply to all the related items".

Reply:
created delete modal
![image](https://github.com/hack4impact-calpoly/wilshire-hope-chest/assets/77358521/d2cb404a-ebe9-4a86-bb57-def2aeca6203)